File: Day1/Java/Main.java
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.util.ArrayList;
import java.util.List;

public class Main {

	public static void main(String[] args) throws IOException {
		File input = new File("input").getAbsoluteFile();
		if (!input.exists()) {
			System.out.println(input.getPath() + " does not exist!");
			System.out.println("trying " + input.getPath() + ".txt instead.");
			input = new File(input.getPath() + ".txt");
			if (!input.exists()) {
				System.err.println("neither " + input.getPath().substring(0, input.getPath().length() - 4) + ",");
				System.err.println("nor " + input.getPath() + " exists!");
				return;
			}
		}

		List<Integer> inputs = new ArrayList<Integer>();
		boolean Double = false;
		boolean Triple = false;
		for(String line:Files.readAllLines(input.toPath())) {
			int nr = Integer.parseInt(line.trim());
			if (!Double && inputs.contains(2020 - nr)) {
				System.out.println("Found matching two number pair!");
				System.out.printf("The numbers are %d and %d, and the result is %d.%n", nr, 2020 - nr, nr * (2020 - nr));
				Double = true;
			}

			if (!Triple) {
				for (int i:inputs) {
					if (inputs.contains(2020 - nr - i)) {
						System.out.println("Found matching three number pair!");
						System.out.printf("The numbers are %d, %d, and %d, and the result is %d.%n", nr, i, 2020 - nr - i, nr * i * (2020 - nr - i));
						Triple = true;
						break;
					}
				}
			}

			inputs.add(nr);
			if (Double && Triple) {
				return;
			}
		}

		System.err.println("Could not find a matching number pair!");
	}
	
}

File: Day1/Python/Main.py
import os

def main():
    input = os.path.dirname(os.getcwd())
    input = os.path.join(input, "input")
    if not os.path.exists(input):
        print(input + " does not exist!")
        input = os.path.join(os.path.dirname(input), "input.txt")
        if not os.path.exists(input):
            print(input + " does not exist!")
            return;

    inputFile = open(input, "r")
    lines = inputFile.readlines()
    inputs = []
    double = False
    triple = False
    for line in lines:
        nr = int(line)
        if not double and 2020 - nr in inputs:
            print("Found matching two number pair!")
            print(f"The numbers are {nr} and {2020 - nr}, and the result is {nr * (2020 - nr)}.")
            double = True

        if not triple:
            for i in inputs:
                if 2020 - nr - i in inputs:
                    print("Found matching three number pair!")
                    print(f"The numbers are {nr}, {i}, and {2020 - nr - i}, and the result is {nr * i * (2020 - nr - i)}.")
                    triple = True
                    break

        inputs.append(nr)
        if double and triple:
            return

    print("Could not find a matching number pair!")

if __name__ == '__main__':
    main()
